ai-safety-debate/confusion_matrix.py

```diff
@@ -0,0 +1,156 @@
+"""
+Computes, saves and shows the confusion matrix as in Figure 3 in AISvD paper
+normalized_confusion_matrix[true_label][liar_label] ... the percentage of liar's wins while arguing for 'liar_label' and the correct label being 'true_label'
+"""
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+from os import remove
+
+from sacred import Experiment
+from sacred.observers import FileStorageObserver
+
+from judge import MNISTJudge, FashionJudge
+from debate import Debate
+from agent import DebateAgent, DebatePlayers
+
+ex = Experiment("confusion matrix")
+ex.observers.append(FileStorageObserver.create("experiments"))
+
+fashion_labels = ['T-shirt/top', 'Trouser', 'Pullover', 'Dress', 'Coat', 'Sandal', 'Shirt', 'Sneaker', 'Bag', 'Ankle boot']
+
+@ex.config
+def cfg():
+    N_to_mask = 4
+    nmbr_of_samples = 20
+    judge_path = None
+    dataset = "mnist"
+    rollouts = 1000
+    index_of_true_agent = 0
+    binary_rewards = True
+    changing_sides = True
+
+
+@ex.automain
+def run(
+    N_to_mask,
+    nmbr_of_samples,
+    judge_path,
+    dataset,
+    rollouts,
+    index_of_true_agent,
+    binary_rewards,
+    changing_sides,
+):
+    """
+    Parameters as always.
+    """
+    # Parse parameters
+    if judge_path:
+        path = judge_path
+    elif dataset:
+        path = "saved_models/" + dataset + str(N_to_mask)
+    else:
+        raise Exception("dataset must be specified")
+
+    if dataset == "mnist":
+        judge = MNISTJudge(
+            N_to_mask=N_to_mask, model_dir=path, binary_rewards=binary_rewards
+        )
+    elif dataset == "fashion":
+        judge = FashionJudge(
+            N_to_mask=N_to_mask, model_dir=path, binary_rewards=binary_rewards
+        )
+    else:
+        raise Exception("Unknown dataset in " + "dataset.txt: " + dataset)
+
+    index_of_lying_agent = (index_of_true_agent + 1)%2
+    
+    # confusion_matrix_counter[label][lying_agent_label] = number of times the lying agent won 
+    # by arguing for the 'lying_agent_label' while the true label being "label"
+    confusion_matrix_counter = np.array([[0 for i in range(10)] for i in range(10)], dtype='f')
+
+    # counts the types of labels of samples
+    labels_frequency = [0 for i in range(10)]
+
+    # play a debate game for each sample and lying label
+    for sample_id in range(nmbr_of_samples):
```

Comment:
This part also seems the same as in the other evaluation script. Can we put this into a function?